text_mining.py

```diff
@@ -0,0 +1,176 @@
+"""
+SoftDes project 3: text mining
+
+Noah Rivkin
+
+use text mining and markov chains to generate text based on silly
+book/movie/comic villain monologues....
+
+or if I cannot find a large sample size imitate Shakespeare
+"""
+
+import requests
+import re
+import pickle
+import random
+import os
+import string
+
+
+suffix_dict_dict = {}
+
+
+def get_text(file_name, url = ''):
+    """
+    opens the file with text in it, if the file exists.
+    If the file does not exist gets the text from the web adress, and
+    stores it in a pickled format,before opening it to read
+    for example http://www.gutenberg.org/cache/epub/100/pg100.txt,
+    which is the complete works of Shakespeare
+    """
+    if os.path.exists(file_name): # checks if the file has already been downloaded
+        f = open(file_name, 'rb')
+        text = pickle.load(f)
+        f.close
+        return text[900: len(text)]
+    else:
+        text = requests.get(url).text
+        f = open(file_name, 'wb')
+        pickle.dump(text, f)
+        f.close
+        return text[900: len(text)]
+
+
+# def histogram(text):
+#     """
+#     generates dictionary with word frequencies
+# 
+#     This function is not used in the main program, and was created for testing purposes
+# 
+#     >>> histogram('d d d d d')
+#     {'d': 5}
+#     """
+#     hist = {}
+#     text = text.replace('\r','\r ') # keeps the return, but still splits up the words
+#     text = text.split(' ')
+#     for word in text:
+#         word = word.strip(string.punctuation + string.whitespace + '1234567890?')
+#         word = word.lower()
+#         if word not in hist and word != '':
+#             hist[word] = 1
+#         elif word != '':
+#             hist[word] = hist[word] + 1
+#     return hist
+# 
+# 
+# def get_most_freq(hist, n):
+#     """
+#     finds the n most common words
+# 
+#     This function is not used in the main program, and was created for testing purposes
+# 
+#     >>> get_most_freq({'a': 5, 's': 3, 'd': 7},2)
+#     [['d', 'a'], [7, 5]]
+#     """
+#     top_n_words = []
+#     top_n_freqs = []
+#     for i in range(n):
+#         most_common = ''
+#         most_times = 0
+#         total = 0
+#         for key in hist:
+#             if key not in top_n_words:
+#                 num = hist[key]
+#                 if num > most_times:
+#                     most_times = num
+#                     most_common = key
+#         top_n_words.append(most_common)
+#         top_n_freqs.append(most_times)
+#     return [top_n_words, top_n_freqs]
```

Comment:
You should delete unnecessary, commented out part of your code that you used during development but is not a part of final code.